libs/MeshKernel/include/MeshKernel/CurvilinearGrid/SnapGridToLandBoundary.hpp

```diff
@@ -0,0 +1,70 @@
+//---- GPL ---------------------------------------------------------------------
+//
+// Copyright (C)  Stichting Deltares, 2011-2024.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "MeshKernel/CurvilinearGrid/CurvilinearGrid.hpp"
+#include "MeshKernel/CurvilinearGrid/CurvilinearGridSnapping.hpp"
+#include "MeshKernel/Definitions.hpp"
+#include "MeshKernel/LandBoundary.hpp"
+#include "MeshKernel/Point.hpp"
+
+namespace meshkernel
+{
+
+    /// @brief Smoothly snap the grid to a land boundary.
+    class SnapGridToLandBoundary : public CurvilinearGridSnapping
```

Comment:
All files in the Curvilinear grid folder start with curvilinear. I am not sure it is the best practice, but you might consider following the same convention also for these files